certbot/certbot/util.py

```diff
@@ -48,6 +49,82 @@ class CSR(NamedTuple):
     form: str
 
 
+class LooseVersion:
+    """A version with loose rules, i.e. any given string is a valid version number.
+
+    but regular comparison is not supported. Instead, the `try_risky_comparison` method is
+    provided, which may return an error if two LooseVersions are 'incomparible'.
+    For example when integer and string version components are present in the same position.
+
+    Differences with old distutils.version.LooseVersion:
+    (https://github.com/python/cpython/blob/v3.10.0/Lib/distutils/version.py#L269)
+    Most version comparisons should give the same result. However, if a version has multiple
+    trailing zeroes, not all of them are used in the comparison. This ensure that, for example,
+    "2.0" and "2.0.0" are equal.
+    """
+
+    def __init__(self, version_string: str) -> None:
+        """Parses a version string into its components.
+
+        :param str version_string: version string
+        """
+        components: List[Union[int, str]]
+        components = [x for x in _VERSION_COMPONENT_RE.split(version_string)
+                              if x and x != '.']
+        for i, obj in enumerate(components):
+            try:
+                components[i] = int(obj)
+            except ValueError:
+                pass
+
+        self.version_components = components
+
+    def try_risky_comparison(self, other: Any) -> int:
```

Comment:
I think this change helps by giving mypy errors (in addition to the runtime errors) if this function is called with the wrong type.
```suggestion
    def try_risky_comparison(self, other: 'LooseVersion') -> int:
```